asapdiscovery-modeling/asapdiscovery/modeling/workflows/protein_prep.py

```diff
@@ -249,6 +231,35 @@ def protein_prep_workflow(inputs: ProteinPrepInputs):
         target_kwargs={"target_name": "ref"},
         ligand_kwargs={"compound_name": "ref_ligand"},
     )
+
+    # setup the path to cached structures
+    cache_path = output_dir / inputs.gen_cache
+
+    # load the cache
```

Comment:
Is it possible to fold the cache checking logic inside the ProteinPrep class or related, the reason this is important is so that other workflows that read cache OR prep, eg large or small scale docking don't have to duplicate this validation logic. I could be misunderstanding how this works though.